sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -119,15 +120,17 @@ __ESIMD_API SurfaceIndex get_surface_index(AccessorTy acc) {
 /// @tparam N Number of elements to read; can be \c 1, \c 2, \c 4, \c 8, \c 16
 ///   or \c 32.
 /// @param p The base address.
-/// @param offsets the vector of 32-bit offsets in bytes. For each lane \c i,
-///   ((byte*)p + offsets[i]) must be element size aligned.
+/// @param offsets the vector of 32-bit or 64-bit offsets in bytes. For each
+/// lane \c i,   ((byte*)p + offsets[i]) must be element size aligned.
 /// @param mask The access mask, defaults to all 1s.
 /// @return A vector of elements read. Elements in masked out lanes are
 ///   undefined.
 ///
-template <typename Tx, int N, class T = detail::__raw_t<Tx>>
-__ESIMD_API std::enable_if_t<detail::isPowerOf2(N, 32), simd<Tx, N>>
-gather(const Tx *p, simd<uint32_t, N> offsets, simd_mask<N> mask = 1) {
+template <typename Tx, int N, class T = detail::__raw_t<Tx>, typename Toffset>
```

Comment:
Parameters with default values (`T`) should go after parameters w/o default values.